custom_components/waste_collection_schedule/waste_collection_schedule/source/bsr_de.py

```diff
@@ -1,130 +1,75 @@
-        "abf_strasse": "Bahnhofstr., 12159 Berlin (Tempelhof-Schöneberg)",
-        "abf_hausnr": 1,
+    "Hufeland_45a": {
+        "schedule_id": "04901100010300413840045A",
     },
-    "Am Ried, 13467 Berlin (Reinickendorf)": {
-        "abf_strasse": "Am Ried, 13467 Berlin (Reinickendorf)",
-        "abf_hausnr": "11G",
+    "Marktstr_1": {
+        "schedule_id": "049011000105000297900010",
     },
 }
 
-
-def initializeSession(abf_strasse, abf_hausnr):
-    s = requests.Session()
-
-    s.get("https://www.bsr.de/abfuhrkalender-20520.php")
-
-    # # start search using street name (without PLZ)
-    args = {"script": "dynamic_search", "step": 1, "q": abf_strasse.split(",")[0]}
-    s.get("https://www.bsr.de/abfuhrkalender_ajax.php", params=args)
-
-    # retrieve house number list
-    args = {"script": "dynamic_search", "step": 2, "q": abf_strasse}
-    s.get("https://www.bsr.de/abfuhrkalender_ajax.php", params=args)
-
-    return s
-
-
-def downloadMonthlyICS(s, abf_strasse, abf_hausnr, month, year):
-    args = {
-        "abf_strasse": abf_strasse.split(",")[0],
-        "abf_hausnr": abf_hausnr,
-        "tab_control": "Monat",
-        "abf_config_weihnachtsbaeume": "",
-        "abf_config_restmuell": "on",
-        "abf_config_biogut": "on",
-        "abf_config_wertstoffe": "on",
-        "abf_config_laubtonne": "on",
-        "abf_selectmonth": f"{month} {year}",
-    }
-    s.post(
-        "https://www.bsr.de/abfuhrkalender_ajax.php?script=dynamic_kalender_ajax",
-        data=args,
-    )
-
-    args["script"] = "dynamic_iCal_ajax"
-    args["abf_strasse"] = abf_strasse
-    r = s.get("https://www.bsr.de/abfuhrkalender_ajax.php", params=args)
-    return r.text
-
-
-def downloadChristmastreeICS(s, abf_strasse, abf_hausnr):
-    args = {
-        "abf_strasse": abf_strasse.split(",")[0],
-        "abf_hausnr": abf_hausnr,
-        "tab_control": "Liste",
-        "abf_config_weihnachtsbaeume": "on",
-        "abf_config_restmuell": "",
-        "abf_config_biogut": "",
-        "abf_config_wertstoffe": "",
-        "abf_config_laubtonne": "",
-    }
-
-    s.post(
-        "https://www.bsr.de/abfuhrkalender_ajax.php?script=dynamic_kalender_ajax",
-        data=args,
-    )
-
-    args["script"] = "dynamic_iCal_ajax"
-    args["abf_strasse"] = abf_strasse
-    r = s.get("https://www.bsr.de/abfuhrkalender_ajax.php", params=args)
-    return r.text
+ENDPOINT_PICKUPS = "https://umnewforms.bsr.de/p/de.bsr.adressen.app/abfuhrEvents"
+FILTERTEMPLATE_PICKUPS = \
+    "AddrKey eq '{id}' and " + \
+    "DateFrom eq datetime'{year_from}-{month:02d}-01T00:00:00' and " + \
+    "DateTo eq datetime'{year_to}-{month:02d}-01T00:00:00'"
+
+WasteInfo = namedtuple("WasteInfo", ["text", "icon"])
+WASTE_CATEGORY_MAP = {
+    "BI": WasteInfo("Biogut", "mdi:bio"),
+    "HM": WasteInfo("Hausmüll", "mdi:trash-can"),
+    "LT": WasteInfo("Laubtonne", "mdi:leaf"),
+    "WS": WasteInfo("Wertstoffe", "mdi:recycle"),
+}
```

Comment:
If you use dataclass you can also have type hints
```suggestion
@dataclass(frozen=True)
class WasteInfo:
    text: str
    icon: str

WASTE_CATEGORY_MAP: dict[str, WasteInfo] = {
    "BI": WasteInfo("Biogut", "mdi:bio"),
    "HM": WasteInfo("Hausmüll", "mdi:trash-can"),
    "LT": WasteInfo("Laubtonne", "mdi:leaf"),
    "WS": WasteInfo("Wertstoffe", "mdi:recycle"),
}
```